src/plot_settings.py

```diff
@@ -126,5 +138,21 @@ def on_close(self, _, parent):
             self.use_custom_plot_style.get_enable_expansion()
         plot_settings.custom_plot_style = \
             self.custom_plot_style.get_selected_item().get_string()
+
+        # Set color cycle
+        if parent.preferences.config["override_style_change"]:
+            parent.canvas = Canvas(parent=parent)
```

Comment:
maybe move this inside the next if statement, so we don't recreate it unnecessarily.